internal/expr/expr_test.go

```diff
@@ -411,6 +411,18 @@ comment */
 string
 of three lines' AND id = $Person.*`,
 		err: `cannot parse expression: line 3, column 26: asterisk not allowed in input expression "$Person.*"`,
+	}, {
+		query: "SELECT &S[:] FROM t",
+		err:   `cannot parse expression: column 8: cannot use slice syntax "S[:]" in output expression`,
+	}, {
+		query: "SELECT &S[0] FROM t",
+		err:   `cannot parse expression: column 8: cannot use slice syntax "S[0]" in output expression`,
```

Comment:
Is there any need for this case? It is testing an invalid form of slice syntax in an invalid context. The other tests should already cover this case.

Reply:
It is a valid form of slice syntax, just the index kind instead of the range one. I wanted to include one example for each of the broad categories.

Reply:
I think these two are the only error messages we can trigger for output expressions, the rest are already tested in the ParseErrors bit, at least until we add support for slices in prepare/query.